native/src/main/java/io/ballerina/stdlib/websocket/WebSocketResourceDispatcher.java

```diff
@@ -542,24 +567,21 @@ private static void handleDataBindingError(WebSocketConnectionInfo connectionInf
         }
     }
 
-    private static String getCustomRemoteMethodName(WebSocketTextMessage textMessage,
-             WebSocketConnection webSocketConnection, WebSocketConnectionInfo.StringAggregator stringAggregator,
-             String dispatchingKey) {
-        String methodName = null;
-        if (null != dispatchingKey) {
-            boolean finalFragment = textMessage.isFinalFragment();
-            if (finalFragment) {
-                stringAggregator.appendAggregateString(textMessage.getText());
-            } else {
-                stringAggregator.appendAggregateString(textMessage.getText());
-                webSocketConnection.readNextFrame();
-            }
+    private static String getCustomRemoteMethodName(String dispatchingKey,
+                                                    WebSocketConnectionInfo.StringAggregator stringAggregator) {
+        if (null == dispatchingKey) {
+            return null;
+        }
+        try {
             BString dispatchingValue = ((BMap) FromJsonString.fromJsonString(StringUtils
                     .fromString(stringAggregator.getAggregateString())))
                     .getStringValue(StringUtils.fromString(dispatchingKey));
-            methodName = createCustomRemoteFunction(dispatchingValue.getValue());
+            return createCustomRemoteFunction(dispatchingValue.getValue());
+        // If any runtime error occurs while retrieving the dispatching value, the default method(onMessage)
+        // name will be used.
+        } catch (RuntimeException e) {
+            return null;
```

Comment:
Related to this comment [1] 

[1] - https://github.com/ballerina-platform/module-ballerina-websocket/pull/1356/files#r1656468643

Reply:
Fixed in [affa6f8](https://github.com/ballerina-platform/module-ballerina-websocket/pull/1356/commits/affa6f834df93e94b88ce5252af1d82b17e7fbad)